## Imports and Drive mount

In [ ]:
# from google.colab import drive
# drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
# Installing Ultralytics
%pip install ultralytics

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.0/1.0 MB 19.3 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 363.4/363.4 MB 4.3 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 13.8/13.8 MB 124.8 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 24.6/24.6 MB 98.4 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 883.7/883.7 kB 60.9 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 664.8/664.8 MB 1.7 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 211.5/211.5 MB 12.1 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 56.3/56.3 MB 44.1 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 127.9/127.9 MB 20.2 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 207.5/207.5 MB 4.9 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 21.1/21.1 MB 104.1 MB/s eta 0:00:00
  Attempting uninstall: nvidia-nvjitlink-cu12
    Found existing installation: nvidia-nvjitlink-cu12 12.5.82
    Uninstall

## Training Base Model

Training the base model which is yolo11m, on 50 epochs using visdrone dataset

In [ ]:
from ultralytics import YOLO

# Training the base model
model = YOLO("yolo11m.pt")

Creating new Ultralytics Settings v0.0.6 file ✅ 
View Ultralytics Settings with 'yolo settings' or at '/root/.config/Ultralytics/settings.json'
Update Settings with 'yolo settings key=value', i.e. 'yolo settings runs_dir=path/to/dir'. For help see https://docs.ultralytics.com/quickstart/#ultralytics-settings.


100%|██████████| 38.8M/38.8M [00:00<00:00, 137MB/s]


In [ ]:
model.train(
    data="visdrone.yaml",
    epochs=50,
    batch=32,
    imgsz=640,
    project="/content/visdrone_yolo11",
    device="cuda:0",
    name="py-expl"
)

Ultralytics 8.3.127 🚀 Python-3.11.12 torch-2.6.0+cu124 CUDA:0 (NVIDIA A100-SXM4-40GB, 40507MiB)
engine/trainer: agnostic_nms=False, amp=True, augment=False, auto_augment=randaugment, batch=32, bgr=0.0, box=7.5, cache=False, cfg=None, classes=None, close_mosaic=10, cls=0.5, conf=None, copy_paste=0.0, copy_paste_mode=flip, cos_lr=False, cutmix=0.0, data=visdrone.yaml, degrees=0.0, deterministic=True, device=cuda:0, dfl=1.5, dnn=False, dropout=0.0, dynamic=False, embed=None, epochs=50, erasing=0.4, exist_ok=False, fliplr=0.5, flipud=0.0, format=torchscript, fraction=1.0, freeze=None, half=False, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, imgsz=640, int8=False, iou=0.7, keras=False, kobj=1.0, line_width=None, lr0=0.01, lrf=0.01, mask_ratio=4, max_det=300, mixup=0.0, mode=train, model=yolo11m.pt, momentum=0.937, mosaic=1.0, multi_scale=False, name=py-expl, nbs=64, nms=False, opset=None, optimize=False, optimizer=auto, overlap_mask=True, patience=100, perspective=0.0, plots=True, pose=12.0, pretrain

100%|██████████| 755k/755k [00:00<00:00, 13.9MB/s]

Overriding model.yaml nc=80 with nc=10

                   from  n    params  module                                       arguments                     
  0                  -1  1      1856  ultralytics.nn.modules.conv.Conv             [3, 64, 3, 2]                 
  1                  -1  1     73984  ultralytics.nn.modules.conv.Conv             [64, 128, 3, 2]               
  2                  -1  1    111872  ultralytics.nn.modules.block.C3k2            [128, 256, 1, True, 0.25]     
  3                  -1  1    590336  ultralytics.nn.modules.conv.Conv             [256, 256, 3, 2]              
  4                  -1  1    444928  ultralytics.nn.modules.block.C3k2            [256, 512, 1, True, 0.25]     
  5                  -1  1   2360320  ultralytics.nn.modules.conv.Conv             [512, 512, 3, 2]              
  6                  -1  1   1380352  ultralytics.nn.modules.block.C3k2            [512, 512, 1, True]           
  7                  -1  1   2360320  ultralytic

 10                  -1  1    990976  ultralytics.nn.modules.block.C2PSA           [512, 512, 1]                 
 11                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          
 12             [-1, 6]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 13                  -1  1   1642496  ultralytics.nn.modules.block.C3k2            [1024, 512, 1, True]          
 14                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          
 15             [-1, 4]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 16                  -1  1    542720  ultralytics.nn.modules.block.C3k2            [1024, 256, 1, True]          
 17                  -1  1    590336  ultralytics.nn.modules.conv.Conv             [256, 256, 3, 2]              
 18            [-1, 13]  1         0  ultralytics.nn.modules.conv.Concat           [1]  

100%|██████████| 5.35M/5.35M [00:00<00:00, 63.9MB/s]


AMP: checks passed ✅
train: Fast image access ✅ (ping: 0.4±0.0 ms, read: 0.8±0.6 MB/s, size: 259.1 KB)


train: Scanning /content/drive/MyDrive/Colab Notebooks/data/visdrone_raw/train/labels... 6449 images, 52 backgrounds, 0 corrupt: 100%|██████████| 6501/6501 [05:46<00:00, 18.74it/s]

train: /content/drive/MyDrive/Colab Notebooks/data/visdrone_raw/train/images/0000137_02220_d_0000163.jpg: 1 duplicate labels removed
train: /content/drive/MyDrive/Colab Notebooks/data/visdrone_raw/train/images/0000140_00118_d_0000002.jpg: 1 duplicate labels removed
train: /content/drive/MyDrive/Colab Notebooks/data/visdrone_raw/train/images/9999945_00000_d_0000114.jpg: 1 duplicate labels removed
train: /content/drive/MyDrive/Colab Notebooks/data/visdrone_raw/train/images/9999987_00000_d_0000049.jpg: 1 duplicate labels removed


train: New cache created: /content/drive/MyDrive/Colab Notebooks/data/visdrone_raw/train/labels.cache
albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01, method='weighted_average', num_output_channels=3), CLAHE(p=0.01, clip_limit=(1.0, 4.0), tile_grid_size=(8, 8))
val: Fast image access ✅ (ping: 0.6±0.1 ms, read: 0.4±0.1 MB/s, size: 131.6 KB)


val: Scanning /content/drive/MyDrive/Colab Notebooks/data/visdrone_raw/val/labels... 548 images, 0 backgrounds, 0 corrupt: 100%|██████████| 548/548 [00:36<00:00, 14.84it/s]


val: New cache created: /content/drive/MyDrive/Colab Notebooks/data/visdrone_raw/val/labels.cache
Plotting labels to /content/visdrone_yolo11/py-expl/labels.jpg... 
optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically... 
optimizer: AdamW(lr=0.000714, momentum=0.9) with parameter groups 106 weight(decay=0.0), 113 weight(decay=0.0005), 112 bias(decay=0.0)
Image sizes 640 train, 640 val
Using 8 dataloader workers
Logging results to /content/visdrone_yolo11/py-expl
Starting training for 50 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       1/50      28.2G       1.41      1.498     0.9566        330        640: 100%|██████████| 204/204 [01:01<00:00,  3.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:04<00:00,  1.96it/s]

                   all        548      38759      0.343      0.264      0.241      0.139



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       2/50      22.3G      1.376      1.108     0.9392        300        640: 100%|██████████| 204/204 [00:57<00:00,  3.56it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.43it/s]

                   all        548      38759       0.39      0.324      0.298      0.173



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       3/50      28.6G      1.387      1.089     0.9371        251        640: 100%|██████████| 204/204 [00:55<00:00,  3.67it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.38it/s]

                   all        548      38759      0.391      0.318      0.294      0.166



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       4/50      26.2G      1.382      1.083      0.939        313        640: 100%|██████████| 204/204 [00:55<00:00,  3.67it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.45it/s]

                   all        548      38759      0.425       0.31      0.303      0.171



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       5/50      23.8G      1.355      1.035     0.9315        321        640: 100%|██████████| 204/204 [00:55<00:00,  3.66it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.46it/s]

                   all        548      38759      0.441      0.327       0.32      0.186



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       6/50      21.6G      1.337      1.001     0.9218        599        640: 100%|██████████| 204/204 [00:55<00:00,  3.66it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.51it/s]

                   all        548      38759       0.47      0.341      0.343      0.201



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       7/50      25.2G      1.312     0.9814     0.9179        457        640: 100%|██████████| 204/204 [00:55<00:00,  3.67it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.46it/s]

                   all        548      38759      0.476      0.357      0.357      0.212



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       8/50      25.8G      1.293     0.9591     0.9163        479        640: 100%|██████████| 204/204 [00:55<00:00,  3.68it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.61it/s]

                   all        548      38759      0.494      0.348      0.365      0.218



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       9/50      22.7G      1.293     0.9502     0.9147        380        640: 100%|██████████| 204/204 [00:55<00:00,  3.67it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.62it/s]

                   all        548      38759      0.443       0.36      0.358      0.212



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      10/50      28.2G      1.277      0.927     0.9081        551        640: 100%|██████████| 204/204 [00:55<00:00,  3.69it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.65it/s]

                   all        548      38759      0.478      0.367      0.373      0.223



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      11/50      25.8G      1.256     0.9129     0.9041        168        640: 100%|██████████| 204/204 [00:55<00:00,  3.67it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.61it/s]

                   all        548      38759      0.479      0.375      0.379      0.225



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      12/50      27.6G       1.26     0.9037     0.9032        742        640: 100%|██████████| 204/204 [00:55<00:00,  3.67it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.54it/s]

                   all        548      38759      0.494      0.383      0.384      0.229



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      13/50      20.8G      1.243     0.8837     0.9011        449        640: 100%|██████████| 204/204 [00:55<00:00,  3.67it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.59it/s]

                   all        548      38759      0.519      0.371      0.392      0.235



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      14/50      21.8G      1.238     0.8805     0.8999        560        640: 100%|██████████| 204/204 [00:55<00:00,  3.68it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.58it/s]

                   all        548      38759      0.506      0.378      0.392      0.234



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      15/50      26.6G      1.231     0.8705      0.897        283        640: 100%|██████████| 204/204 [00:55<00:00,  3.67it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.65it/s]

                   all        548      38759      0.511      0.389      0.404      0.245



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      16/50      24.4G      1.223     0.8529     0.8947        529        640: 100%|██████████| 204/204 [00:55<00:00,  3.67it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.64it/s]

                   all        548      38759      0.526       0.39      0.408      0.246



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      17/50      29.5G      1.218      0.855     0.8948        350        640: 100%|██████████| 204/204 [00:55<00:00,  3.67it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.67it/s]

                   all        548      38759      0.512      0.397      0.406      0.247



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      18/50        24G      1.204     0.8378     0.8917        271        640: 100%|██████████| 204/204 [00:55<00:00,  3.67it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.58it/s]

                   all        548      38759      0.532       0.39      0.411      0.249



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      19/50      24.8G      1.199     0.8235     0.8891        440        640: 100%|██████████| 204/204 [00:55<00:00,  3.67it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.48it/s]

                   all        548      38759      0.523      0.401      0.415      0.251



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      20/50      23.2G      1.193     0.8244     0.8885        380        640: 100%|██████████| 204/204 [00:55<00:00,  3.68it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.60it/s]

                   all        548      38759      0.531      0.398      0.416      0.249



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      21/50      24.9G      1.191     0.8188     0.8897        379        640: 100%|██████████| 204/204 [00:55<00:00,  3.68it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.62it/s]

                   all        548      38759      0.537      0.417      0.428      0.258



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      22/50      30.4G      1.188     0.8063     0.8891        715        640: 100%|██████████| 204/204 [00:55<00:00,  3.68it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.69it/s]

                   all        548      38759      0.545      0.403      0.429      0.262



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      23/50        25G      1.176     0.7988     0.8832        335        640: 100%|██████████| 204/204 [00:55<00:00,  3.66it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.67it/s]

                   all        548      38759      0.532      0.406      0.425      0.258



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      24/50      24.8G      1.172     0.7907     0.8827        325        640: 100%|██████████| 204/204 [00:55<00:00,  3.68it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.62it/s]

                   all        548      38759       0.54      0.417      0.432      0.264



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      25/50      29.2G      1.158     0.7819     0.8816        464        640: 100%|██████████| 204/204 [00:55<00:00,  3.67it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.64it/s]

                   all        548      38759      0.536      0.408      0.423      0.257



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      26/50      24.7G      1.164     0.7752     0.8824        241        640: 100%|██████████| 204/204 [00:55<00:00,  3.66it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.63it/s]

                   all        548      38759      0.554      0.413      0.435      0.263



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      27/50      23.5G      1.149     0.7653     0.8785        302        640: 100%|██████████| 204/204 [00:55<00:00,  3.69it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.57it/s]

                   all        548      38759      0.537      0.418      0.433      0.268



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      28/50      24.9G      1.163     0.7692     0.8811        230        640: 100%|██████████| 204/204 [00:55<00:00,  3.68it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.65it/s]

                   all        548      38759      0.536       0.41      0.433      0.266



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      29/50      25.5G      1.151     0.7651     0.8782        371        640: 100%|██████████| 204/204 [00:55<00:00,  3.67it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.67it/s]

                   all        548      38759      0.551      0.414      0.438      0.269



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      30/50      25.4G      1.148     0.7525     0.8763        522        640: 100%|██████████| 204/204 [00:55<00:00,  3.66it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.70it/s]

                   all        548      38759      0.546      0.425      0.442      0.272



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      31/50      23.8G      1.142     0.7477      0.876        324        640: 100%|██████████| 204/204 [00:55<00:00,  3.66it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.64it/s]

                   all        548      38759      0.552      0.423      0.443      0.272



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      32/50      21.8G      1.127     0.7346     0.8732        430        640: 100%|██████████| 204/204 [00:55<00:00,  3.67it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.60it/s]

                   all        548      38759      0.536      0.426      0.443      0.272



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      33/50      28.3G      1.123     0.7276     0.8736        312        640: 100%|██████████| 204/204 [00:55<00:00,  3.67it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.65it/s]

                   all        548      38759      0.545      0.422      0.445      0.272



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      34/50      27.1G      1.121     0.7258     0.8718        348        640: 100%|██████████| 204/204 [00:55<00:00,  3.67it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.66it/s]

                   all        548      38759      0.554      0.421      0.446      0.273



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      35/50      22.6G      1.112      0.715       0.87        408        640: 100%|██████████| 204/204 [00:55<00:00,  3.67it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.73it/s]

                   all        548      38759      0.554      0.427      0.449      0.275



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      36/50        22G      1.114     0.7142     0.8701        399        640: 100%|██████████| 204/204 [00:55<00:00,  3.68it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.70it/s]

                   all        548      38759      0.553      0.425      0.449      0.274



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      37/50      27.2G      1.108      0.706     0.8694        310        640: 100%|██████████| 204/204 [00:55<00:00,  3.67it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.61it/s]

                   all        548      38759      0.559      0.421      0.448      0.277



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      38/50      30.5G      1.104     0.6991     0.8685        197        640: 100%|██████████| 204/204 [00:55<00:00,  3.67it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.56it/s]

                   all        548      38759      0.553      0.428      0.448      0.275



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      39/50        31G      1.105     0.6999     0.8671        437        640: 100%|██████████| 204/204 [00:55<00:00,  3.67it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.68it/s]

                   all        548      38759      0.557      0.426      0.449      0.277



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      40/50      28.4G      1.104     0.6943     0.8689        393        640: 100%|██████████| 204/204 [00:55<00:00,  3.67it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.59it/s]

                   all        548      38759      0.546      0.436      0.451      0.277


Closing dataloader mosaic
albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01, method='weighted_average', num_output_channels=3), CLAHE(p=0.01, clip_limit=(1.0, 4.0), tile_grid_size=(8, 8))

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      41/50      25.8G      1.087     0.6818     0.8691        241        640: 100%|██████████| 204/204 [00:56<00:00,  3.61it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.70it/s]

                   all        548      38759      0.567      0.421      0.445      0.274



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      42/50      24.2G       1.08     0.6686     0.8671        446        640: 100%|██████████| 204/204 [00:53<00:00,  3.78it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.66it/s]

                   all        548      38759      0.556      0.426      0.447      0.275



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      43/50      24.7G      1.069      0.658      0.865         85        640: 100%|██████████| 204/204 [00:53<00:00,  3.78it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.63it/s]

                   all        548      38759      0.557       0.43      0.449      0.276



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      44/50      23.7G      1.067     0.6506     0.8644        154        640: 100%|██████████| 204/204 [00:53<00:00,  3.78it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.63it/s]

                   all        548      38759      0.562      0.422      0.448      0.275



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      45/50      24.4G      1.064     0.6459     0.8642        225        640: 100%|██████████| 204/204 [00:53<00:00,  3.78it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.71it/s]

                   all        548      38759      0.556      0.427      0.452      0.279



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      46/50      22.1G       1.05     0.6335     0.8609        302        640: 100%|██████████| 204/204 [00:53<00:00,  3.78it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.72it/s]

                   all        548      38759      0.568       0.43      0.453      0.278



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      47/50      20.4G      1.048      0.628     0.8614        208        640: 100%|██████████| 204/204 [00:54<00:00,  3.78it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.58it/s]

                   all        548      38759      0.573      0.422      0.451      0.278



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      48/50      25.6G      1.045     0.6253     0.8592        216        640: 100%|██████████| 204/204 [00:54<00:00,  3.77it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.72it/s]

                   all        548      38759      0.571      0.429      0.454       0.28



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      49/50      26.2G      1.043     0.6202     0.8601        120        640: 100%|██████████| 204/204 [00:53<00:00,  3.78it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.72it/s]

                   all        548      38759      0.559      0.435      0.453      0.279



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      50/50      26.6G      1.032     0.6111     0.8578        110        640: 100%|██████████| 204/204 [00:53<00:00,  3.78it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:03<00:00,  2.61it/s]

                   all        548      38759      0.563      0.432      0.454      0.279



50 epochs completed in 0.834 hours.
Optimizer stripped from /content/visdrone_yolo11/py-expl/weights/last.pt, 40.5MB
Optimizer stripped from /content/visdrone_yolo11/py-expl/weights/best.pt, 40.5MB

Validating /content/visdrone_yolo11/py-expl/weights/best.pt...
Ultralytics 8.3.127 🚀 Python-3.11.12 torch-2.6.0+cu124 CUDA:0 (NVIDIA A100-SXM4-40GB, 40507MiB)
YOLO11m summary (fused): 125 layers, 20,037,742 parameters, 0 gradients, 67.7 GFLOPs


                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 9/9 [00:10<00:00,  1.21s/it]


                   all        548      38759       0.57       0.43      0.454       0.28
            pedestrian        520       8844      0.621      0.431      0.493      0.236
                people        482       5125      0.652      0.296      0.378      0.153
               bicycle        364       1287      0.297      0.258      0.189     0.0881
                   car        515      14064      0.784      0.792      0.826      0.602
                   van        421       1975      0.558      0.489      0.498      0.361
                 truck        266        750      0.602       0.44      0.458      0.315
              tricycle        337       1045      0.501      0.334      0.339      0.194
       awning-tricycle        220        532      0.356      0.203      0.207      0.135
                   bus        131        251      0.762      0.574      0.656      0.489
                 motor        485       4886      0.564      0.483      0.501      0.233
Speed: 0.1ms preproce

ultralytics.utils.metrics.DetMetrics object with attributes:

ap_class_index: array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
box: ultralytics.utils.metrics.Metric object
confusion_matrix: <ultralytics.utils.metrics.ConfusionMatrix object at 0x7ef766f84050>
curves: ['Precision-Recall(B)', 'F1-Confidence(B)', 'Precision-Confidence(B)', 'Recall-Confidence(B)']
curves_results: [[array([          0,    0.001001,    0.002002,    0.003003,    0.004004,    0.005005,    0.006006,    0.007007,    0.008008,    0.009009,     0.01001,    0.011011,    0.012012,    0.013013,    0.014014,    0.015015,    0.016016,    0.017017,    0.018018,    0.019019,     0.02002,    0.021021,    0.022022,    0.023023,
          0.024024,    0.025025,    0.026026,    0.027027,    0.028028,    0.029029,     0.03003,    0.031031,    0.032032,    0.033033,    0.034034,    0.035035,    0.036036,    0.037037,    0.038038,    0.039039,     0.04004,    0.041041,    0.042042,    0.043043,    0.044044,    0.045045,    0.046046,    0.0

For this model, we get `mAP50 : 0.454` and `mAP50-95 : 0.28` 

---

## Traing Backbone Modified Model

Modified the backbone of the model by adding a Vision Transformer (ViT) stem. The stem consists of:

1. **Patch Embedder**  
   Splits the input image into non-overlapping 16×16 patches and projects each patch into a 64-dimensional token vector via a convolutional patch embedding layer.

2. **Transformer Encoder Layers**  
   Applies two `nn.TransformerEncoderLayer` blocks (with multi-head self-attention and an MLP) to the sequence of patch tokens, enabling the network to learn long-range, context-aware relationships between image regions.

3. **Adapter**  
   Uses a `ConvTranspose2d` layer followed by a SiLU activation to reassemble the attended token embeddings back into a full-resolution, 3-channel feature map that can feed directly into the rest of the YOLO backbone.

In [ ]:
from ultralytics import YOLO

base = YOLO("yolo11m.yaml")

In [ ]:
base

YOLO(
  (model): DetectionModel(
    (model): Sequential(
      (0): Conv(
        (conv): Conv2d(3, 64, kernel_size=(3, 3), stride=(2, 2), padding=(1, 1), bias=False)
        (bn): BatchNorm2d(64, eps=0.001, momentum=0.03, affine=True, track_running_stats=True)
        (act): SiLU(inplace=True)
      )
      (1): Conv(
        (conv): Conv2d(64, 128, kernel_size=(3, 3), stride=(2, 2), padding=(1, 1), bias=False)
        (bn): BatchNorm2d(128, eps=0.001, momentum=0.03, affine=True, track_running_stats=True)
        (act): SiLU(inplace=True)
      )
      (2): C3k2(
        (cv1): Conv(
          (conv): Conv2d(128, 128, kernel_size=(1, 1), stride=(1, 1), bias=False)
          (bn): BatchNorm2d(128, eps=0.001, momentum=0.03, affine=True, track_running_stats=True)
          (act): SiLU(inplace=True)
        )
        (cv2): Conv(
          (conv): Conv2d(192, 256, kernel_size=(1, 1), stride=(1, 1), bias=False)
          (bn): BatchNorm2d(256, eps=0.001, momentum=0.03, affine=True, track_

In [ ]:
import torch.nn as nn

In [ ]:
class ViTStem(nn.Module):
    def __init__(self, patch_size=16, in_chans=3, embed_dim=64, n_layers=2, nhead=4, mlp_ratio=2.0, dropout=0.05):
        super().__init__()
        self.patch_embed = nn.Conv2d(
            in_channels=in_chans,
            out_channels=embed_dim,
            kernel_size=patch_size,
            stride=patch_size
        )

        layers = []
        for _ in range(n_layers):
            layers.append(nn.TransformerEncoderLayer(
                d_model=embed_dim,
                nhead=nhead,
                dim_feedforward=int(embed_dim * mlp_ratio),
                dropout=dropout
            ))
        self.transformers = nn.Sequential(*layers)

        self.adapter = nn.ConvTranspose2d(
            in_channels=embed_dim,
            out_channels=in_chans,
            kernel_size=patch_size,
            stride=patch_size,
            padding=0,
            bias=False
        )

    def forward(self, x):
        x = self.patch_embed(x)
        b,c,h,w = x.shape
        x = x.flatten(2).permute(0,2,1)
        x = self.transformers(x)
        x = x.permute(0, 2, 1).view(b, c, h, w)
        x = nn.functional.SiLU(self.adapter(x))
        return x

In [ ]:
raw = base.model.model
base.model.model = nn.Sequential(ViTStem(), raw)

In [ ]:
base.train(
    data="visdrone.yaml",
    epochs=50,
    batch=48,
    imgsz=640,
    project="/content/visdrone_yolo11",
    device="cuda:0",
    name="py-mod-expl"
)

Ultralytics 8.3.127 🚀 Python-3.11.12 torch-2.6.0+cu124 CUDA:0 (NVIDIA A100-SXM4-40GB, 40507MiB)
engine/trainer: agnostic_nms=False, amp=True, augment=False, auto_augment=randaugment, batch=48, bgr=0.0, box=7.5, cache=False, cfg=None, classes=None, close_mosaic=10, cls=0.5, conf=None, copy_paste=0.0, copy_paste_mode=flip, cos_lr=False, cutmix=0.0, data=visdrone.yaml, degrees=0.0, deterministic=True, device=cuda:0, dfl=1.5, dnn=False, dropout=0.0, dynamic=False, embed=None, epochs=50, erasing=0.4, exist_ok=False, fliplr=0.5, flipud=0.0, format=torchscript, fraction=1.0, freeze=None, half=False, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, imgsz=640, int8=False, iou=0.7, keras=False, kobj=1.0, line_width=None, lr0=0.01, lrf=0.01, mask_ratio=4, max_det=300, mixup=0.0, mode=train, model=yolo11m.yaml, momentum=0.937, mosaic=1.0, multi_scale=False, name=py-mod-expl, nbs=64, nms=False, opset=None, optimize=False, optimizer=auto, overlap_mask=True, patience=100, perspective=0.0, plots=True, pose=12.0, pr

train: Scanning /content/drive/MyDrive/Colab Notebooks/data/visdrone_raw/train/labels.cache... 6449 images, 52 backgrounds, 0 corrupt: 100%|██████████| 6501/6501 [00:00<?, ?it/s]

train: /content/drive/MyDrive/Colab Notebooks/data/visdrone_raw/train/images/0000137_02220_d_0000163.jpg: 1 duplicate labels removed
train: /content/drive/MyDrive/Colab Notebooks/data/visdrone_raw/train/images/0000140_00118_d_0000002.jpg: 1 duplicate labels removed
train: /content/drive/MyDrive/Colab Notebooks/data/visdrone_raw/train/images/9999945_00000_d_0000114.jpg: 1 duplicate labels removed
train: /content/drive/MyDrive/Colab Notebooks/data/visdrone_raw/train/images/9999987_00000_d_0000049.jpg: 1 duplicate labels removed
albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01, method='weighted_average', num_output_channels=3), CLAHE(p=0.01, clip_limit=(1.0, 4.0), tile_grid_size=(8, 8))


val: Fast image access ✅ (ping: 1.1±1.3 ms, read: 58.2±16.2 MB/s, size: 131.6 KB)


val: Scanning /content/drive/MyDrive/Colab Notebooks/data/visdrone_raw/val/labels.cache... 548 images, 0 backgrounds, 0 corrupt: 100%|██████████| 548/548 [00:00<?, ?it/s]


Plotting labels to /content/visdrone_yolo11/py-mod-expl/labels.jpg... 
optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically... 
optimizer: AdamW(lr=0.000714, momentum=0.9) with parameter groups 106 weight(decay=0.0), 113 weight(decay=0.000375), 112 bias(decay=0.0)
Image sizes 640 train, 640 val
Using 8 dataloader workers
Logging results to /content/visdrone_yolo11/py-mod-expl
Starting training for 50 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       1/50      34.2G      4.234      3.982      3.188       1875        640: 100%|██████████| 136/136 [01:45<00:00,  1.28it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.93it/s]

                   all        548      38759     0.0115     0.0609     0.0124      0.004



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       2/50        37G       2.54      2.072      1.585       1748        640: 100%|██████████| 136/136 [00:55<00:00,  2.45it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.57it/s]

                   all        548      38759      0.317      0.116     0.0841     0.0384



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       3/50      36.3G      2.115      1.714      1.288       1765        640: 100%|██████████| 136/136 [00:55<00:00,  2.44it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.62it/s]

                   all        548      38759      0.177      0.173      0.114     0.0549



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       4/50      33.9G      1.906      1.554      1.181       1244        640: 100%|██████████| 136/136 [00:55<00:00,  2.44it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.68it/s]

                   all        548      38759      0.327      0.184      0.154     0.0795



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       5/50      34.8G      1.779      1.425      1.124       1980        640: 100%|██████████| 136/136 [00:55<00:00,  2.45it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.70it/s]

                   all        548      38759      0.253      0.206      0.171     0.0883



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       6/50      30.3G      1.722      1.368      1.094       1521        640: 100%|██████████| 136/136 [00:55<00:00,  2.43it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.74it/s]

                   all        548      38759      0.288      0.225      0.192      0.102



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       7/50      37.2G      1.655      1.309       1.07       1209        640: 100%|██████████| 136/136 [00:55<00:00,  2.45it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.65it/s]

                   all        548      38759      0.322      0.237      0.218      0.117



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       8/50      37.6G      1.605      1.255      1.054       1590        640: 100%|██████████| 136/136 [00:55<00:00,  2.45it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.78it/s]

                   all        548      38759      0.325      0.254      0.221       0.12



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       9/50      34.8G      1.572      1.231      1.043       1320        640: 100%|██████████| 136/136 [00:55<00:00,  2.44it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.68it/s]

                   all        548      38759      0.325      0.242      0.221       0.12



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      10/50      32.2G      1.545      1.196      1.031       1207        640: 100%|██████████| 136/136 [00:55<00:00,  2.43it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.76it/s]

                   all        548      38759      0.358      0.269      0.251      0.137



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      11/50      33.7G      1.555       1.19      1.019       1815        640: 100%|██████████| 136/136 [00:55<00:00,  2.45it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.62it/s]

                   all        548      38759       0.35      0.274      0.254       0.14



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      12/50      34.4G      1.548      1.186      1.011       1658        640: 100%|██████████| 136/136 [00:55<00:00,  2.46it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.69it/s]

                   all        548      38759      0.362      0.274      0.263      0.146



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      13/50      31.7G      1.521      1.168     0.9987       1252        640: 100%|██████████| 136/136 [00:55<00:00,  2.45it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.71it/s]

                   all        548      38759      0.381      0.275       0.27      0.152



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      14/50      28.6G      1.495       1.14     0.9911       1261        640: 100%|██████████| 136/136 [00:55<00:00,  2.45it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.69it/s]

                   all        548      38759      0.398      0.298      0.285       0.16



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      15/50      28.8G      1.489      1.125     0.9861       1814        640: 100%|██████████| 136/136 [00:55<00:00,  2.45it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.68it/s]

                   all        548      38759      0.407      0.313      0.299      0.171



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      16/50      35.5G      1.464      1.103     0.9808       1621        640: 100%|██████████| 136/136 [00:55<00:00,  2.46it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.64it/s]

                   all        548      38759      0.401       0.31      0.299      0.171



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      17/50        31G      1.454      1.101     0.9788       1705        640: 100%|██████████| 136/136 [00:55<00:00,  2.44it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.69it/s]

                   all        548      38759      0.414      0.318      0.308      0.174



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      18/50      37.7G      1.445      1.083     0.9709       1606        640: 100%|██████████| 136/136 [00:55<00:00,  2.46it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.64it/s]

                   all        548      38759      0.399      0.319      0.309      0.177



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      19/50      34.2G      1.443      1.071       0.97       2017        640: 100%|██████████| 136/136 [00:55<00:00,  2.47it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.66it/s]

                   all        548      38759      0.421      0.315      0.312       0.18



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      20/50      35.5G       1.42      1.054     0.9643       1368        640: 100%|██████████| 136/136 [00:55<00:00,  2.45it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.71it/s]

                   all        548      38759      0.423      0.334      0.327      0.189



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      21/50      34.9G      1.407      1.045     0.9634       2206        640: 100%|██████████| 136/136 [00:55<00:00,  2.46it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.71it/s]

                   all        548      38759      0.426      0.331      0.327       0.19



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      22/50      36.2G      1.394       1.03     0.9585       1289        640: 100%|██████████| 136/136 [00:55<00:00,  2.45it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.71it/s]

                   all        548      38759      0.435      0.333       0.33      0.189



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      23/50      36.4G      1.391      1.025     0.9557       2012        640: 100%|██████████| 136/136 [00:55<00:00,  2.45it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.67it/s]

                   all        548      38759      0.434      0.335       0.33      0.191



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      24/50      32.2G      1.383       1.01     0.9545       1598        640: 100%|██████████| 136/136 [00:55<00:00,  2.45it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.70it/s]

                   all        548      38759      0.441      0.334      0.337      0.196



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      25/50      37.8G      1.372      1.001     0.9485       2206        640: 100%|██████████| 136/136 [00:55<00:00,  2.45it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.71it/s]

                   all        548      38759      0.443       0.35      0.342      0.199



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      26/50      38.2G      1.358     0.9904     0.9498       1188        640: 100%|██████████| 136/136 [00:55<00:00,  2.44it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.75it/s]

                   all        548      38759      0.442      0.344      0.343        0.2



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      27/50      36.4G      1.354     0.9868     0.9465       1438        640: 100%|██████████| 136/136 [00:55<00:00,  2.44it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.71it/s]

                   all        548      38759       0.44      0.343      0.343      0.199



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      28/50      30.6G      1.362     0.9878     0.9477       1917        640: 100%|██████████| 136/136 [00:55<00:00,  2.43it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.65it/s]

                   all        548      38759      0.454       0.35      0.351      0.206



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      29/50      35.7G      1.349     0.9755     0.9449       1731        640: 100%|██████████| 136/136 [00:55<00:00,  2.46it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.77it/s]

                   all        548      38759       0.46      0.348      0.354      0.207



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      30/50      32.8G      1.333     0.9585     0.9402       1784        640: 100%|██████████| 136/136 [00:55<00:00,  2.44it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.75it/s]

                   all        548      38759      0.445      0.359      0.353      0.208



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      31/50        37G      1.328     0.9519     0.9379       1310        640: 100%|██████████| 136/136 [00:55<00:00,  2.45it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.72it/s]

                   all        548      38759      0.463       0.35      0.353      0.208



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      32/50      32.7G      1.324     0.9475     0.9383       1347        640: 100%|██████████| 136/136 [00:55<00:00,  2.43it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.70it/s]

                   all        548      38759      0.459      0.359      0.358      0.211



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      33/50      34.7G      1.309     0.9356     0.9352       1751        640: 100%|██████████| 136/136 [00:55<00:00,  2.44it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.77it/s]

                   all        548      38759      0.475      0.357      0.363      0.214



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      34/50      32.2G      1.314     0.9373     0.9346       1294        640: 100%|██████████| 136/136 [00:56<00:00,  2.43it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.75it/s]

                   all        548      38759      0.478      0.347      0.359      0.212



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      35/50      35.9G      1.298     0.9259     0.9318       2091        640: 100%|██████████| 136/136 [00:55<00:00,  2.44it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.77it/s]

                   all        548      38759      0.476      0.363      0.367      0.216



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      36/50      38.6G      1.302     0.9226     0.9322       1611        640: 100%|██████████| 136/136 [00:55<00:00,  2.44it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.70it/s]

                   all        548      38759      0.479      0.363      0.369      0.219



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      37/50      34.2G      1.288     0.9086       0.93       1716        640: 100%|██████████| 136/136 [00:55<00:00,  2.45it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.75it/s]

                   all        548      38759      0.474      0.368      0.369      0.218



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      38/50      32.2G      1.293     0.9168     0.9299       1773        640: 100%|██████████| 136/136 [00:55<00:00,  2.45it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.73it/s]

                   all        548      38759      0.486      0.363      0.371       0.22



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      39/50      32.6G      1.289     0.9104     0.9274       1465        640: 100%|██████████| 136/136 [00:55<00:00,  2.45it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.76it/s]

                   all        548      38759      0.482      0.366      0.373      0.221



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      40/50      37.3G       1.28     0.8989     0.9265       1485        640: 100%|██████████| 136/136 [00:55<00:00,  2.44it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.71it/s]

                   all        548      38759      0.487      0.368      0.377      0.223


Closing dataloader mosaic
albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01, method='weighted_average', num_output_channels=3), CLAHE(p=0.01, clip_limit=(1.0, 4.0), tile_grid_size=(8, 8))

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      41/50      31.3G      1.257     0.8798      0.932        929        640: 100%|██████████| 136/136 [00:56<00:00,  2.40it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.70it/s]

                   all        548      38759      0.482      0.363      0.371       0.22



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      42/50      29.1G      1.241     0.8637     0.9261       1060        640: 100%|██████████| 136/136 [00:53<00:00,  2.53it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.78it/s]

                   all        548      38759      0.492      0.364      0.375      0.223



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      43/50      38.1G      1.238     0.8568     0.9251       1054        640: 100%|██████████| 136/136 [00:53<00:00,  2.53it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.74it/s]

                   all        548      38759      0.484      0.368      0.375      0.223



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      44/50      31.2G      1.231     0.8514     0.9247        846        640: 100%|██████████| 136/136 [00:54<00:00,  2.51it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.78it/s]

                   all        548      38759      0.488      0.365      0.377      0.225



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      45/50        29G      1.229     0.8446     0.9225       1018        640: 100%|██████████| 136/136 [00:53<00:00,  2.52it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.80it/s]

                   all        548      38759      0.486      0.372      0.378      0.225



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      46/50        38G      1.222     0.8379     0.9207        780        640: 100%|██████████| 136/136 [00:53<00:00,  2.53it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.67it/s]

                   all        548      38759      0.487      0.372      0.379      0.226



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      47/50        35G      1.217     0.8321     0.9205        850        640: 100%|██████████| 136/136 [00:54<00:00,  2.51it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.72it/s]

                   all        548      38759      0.489      0.371      0.381      0.227



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      48/50      36.3G      1.214     0.8287     0.9183        819        640: 100%|██████████| 136/136 [00:53<00:00,  2.52it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.74it/s]

                   all        548      38759      0.489      0.372      0.381      0.227



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      49/50      32.3G      1.209     0.8241     0.9171       1072        640: 100%|██████████| 136/136 [00:54<00:00,  2.51it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.74it/s]

                   all        548      38759      0.498       0.37      0.381      0.227



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      50/50      32.8G      1.202     0.8174     0.9162        848        640: 100%|██████████| 136/136 [00:53<00:00,  2.53it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.79it/s]

                   all        548      38759      0.495       0.37      0.381      0.228



50 epochs completed in 0.848 hours.
Optimizer stripped from /content/visdrone_yolo11/py-mod-expl/weights/last.pt, 40.5MB
Optimizer stripped from /content/visdrone_yolo11/py-mod-expl/weights/best.pt, 40.5MB

Validating /content/visdrone_yolo11/py-mod-expl/weights/best.pt...
Ultralytics 8.3.127 🚀 Python-3.11.12 torch-2.6.0+cu124 CUDA:0 (NVIDIA A100-SXM4-40GB, 40507MiB)
YOLO11m summary (fused): 125 layers, 20,037,742 parameters, 0 gradients, 67.7 GFLOPs


                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:12<00:00,  2.02s/it]


                   all        548      38759      0.495       0.37      0.381      0.227
            pedestrian        520       8844      0.518      0.377      0.403      0.181
                people        482       5125      0.559      0.245      0.307      0.113
               bicycle        364       1287      0.247      0.149      0.113     0.0474
                   car        515      14064      0.711      0.764      0.789       0.56
                   van        421       1975      0.498      0.445      0.446      0.316
                 truck        266        750      0.494      0.349      0.355      0.244
              tricycle        337       1045       0.44      0.266      0.261      0.143
       awning-tricycle        220        532      0.313      0.175      0.156      0.096
                   bus        131        251      0.669      0.514      0.566      0.395
                 motor        485       4886      0.505      0.415      0.415       0.18
Speed: 0.1ms preproce

ultralytics.utils.metrics.DetMetrics object with attributes:

ap_class_index: array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
box: ultralytics.utils.metrics.Metric object
confusion_matrix: <ultralytics.utils.metrics.ConfusionMatrix object at 0x7ef703be0850>
curves: ['Precision-Recall(B)', 'F1-Confidence(B)', 'Precision-Confidence(B)', 'Recall-Confidence(B)']
curves_results: [[array([          0,    0.001001,    0.002002,    0.003003,    0.004004,    0.005005,    0.006006,    0.007007,    0.008008,    0.009009,     0.01001,    0.011011,    0.012012,    0.013013,    0.014014,    0.015015,    0.016016,    0.017017,    0.018018,    0.019019,     0.02002,    0.021021,    0.022022,    0.023023,
          0.024024,    0.025025,    0.026026,    0.027027,    0.028028,    0.029029,     0.03003,    0.031031,    0.032032,    0.033033,    0.034034,    0.035035,    0.036036,    0.037037,    0.038038,    0.039039,     0.04004,    0.041041,    0.042042,    0.043043,    0.044044,    0.045045,    0.046046,    0.0

Here we get, `mAP50 : 0.381` and `mAP50-95 : 0.227`

---

## Training by Augmenting Data and changing Hyper-parameters


This model was not a pre-trained yolo11m, it had empty weights, but was trained differently than others.

- **Augmentation Pipeline**  
  - **Standard augment** enabled (`augment=True`) without AutoAugment  
  - **Mosaic** mixing of 4 images per batch (`mosaic=True`)  
  - **MixUp** blending with strength **0.5**  
  - **Color jitter** in HSV space: hue ±0.015, saturation ±0.7, value ±0.4  
  - **Horizontal flip** probability **0.5**, no vertical flips  
  - **Random translation** up to **10%**, scaling up to **50%**, no shear  

**Optimizer & LR Schedule**  
  - **Optimizer**: AdamW with initial LR (`lr0`) of **1e-3**, momentum **0.80**, weight decay **5e-4**  
  - **Cosine LR decay** (`cos_lr=True`) down to a final factor `lrf=1e-3` of the initial LR  
  - **Warmup** for **5 epochs** to ramp up the learning rate smoothly  

In [ ]:
from ultralytics import YOLO

aug_model = YOLO("yolo11m.yaml")

In [ ]:
aug_model.train(
    data="visdrone.yaml",
    epochs=50,
    batch=48,
    imgsz=640,
    project="visdrone_yolo11",
    device="cuda:0",
    name="py-aug-expl",
    optimizer='AdamW',
    lr0=1e-3,
    momentum=0.80,
    weight_decay=5e-4,

    cos_lr=True,
    lrf=1e-3,
    warmup_epochs=5,

    auto_augment=False,
    augment=True,
    mosaic=True,
    mixup=0.5,
    hsv_h=0.015,
    hsv_s=0.7,
    hsv_v=0.4,
    flipud=0.0,
    fliplr=0.5,
    translate=0.1,
    scale=0.5,
    shear=0.0
)

Ultralytics 8.3.127 🚀 Python-3.11.12 torch-2.6.0+cu124 CUDA:0 (NVIDIA A100-SXM4-40GB, 40507MiB)
engine/trainer: agnostic_nms=False, amp=True, augment=True, auto_augment=False, batch=48, bgr=0.0, box=7.5, cache=False, cfg=None, classes=None, close_mosaic=10, cls=0.5, conf=None, copy_paste=0.0, copy_paste_mode=flip, cos_lr=True, cutmix=0.0, data=visdrone.yaml, degrees=0.0, deterministic=True, device=cuda:0, dfl=1.5, dnn=False, dropout=0.0, dynamic=False, embed=None, epochs=50, erasing=0.4, exist_ok=False, fliplr=0.5, flipud=0.0, format=torchscript, fraction=1.0, freeze=None, half=False, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, imgsz=640, int8=False, iou=0.7, keras=False, kobj=1.0, line_width=None, lr0=0.001, lrf=0.001, mask_ratio=4, max_det=300, mixup=0.5, mode=train, model=yolo11m.yaml, momentum=0.8, mosaic=True, multi_scale=False, name=py-aug-expl, nbs=64, nms=False, opset=None, optimize=False, optimizer=AdamW, overlap_mask=True, patience=100, perspective=0.0, plots=True, pose=12.0, pretrain

train: Scanning /content/drive/MyDrive/Colab Notebooks/data/visdrone_raw/train/labels.cache... 6449 images, 52 backgrounds, 0 corrupt: 100%|██████████| 6501/6501 [00:00<?, ?it/s]

train: /content/drive/MyDrive/Colab Notebooks/data/visdrone_raw/train/images/0000137_02220_d_0000163.jpg: 1 duplicate labels removed
train: /content/drive/MyDrive/Colab Notebooks/data/visdrone_raw/train/images/0000140_00118_d_0000002.jpg: 1 duplicate labels removed
train: /content/drive/MyDrive/Colab Notebooks/data/visdrone_raw/train/images/9999945_00000_d_0000114.jpg: 1 duplicate labels removed
train: /content/drive/MyDrive/Colab Notebooks/data/visdrone_raw/train/images/9999987_00000_d_0000049.jpg: 1 duplicate labels removed
albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01, method='weighted_average', num_output_channels=3), CLAHE(p=0.01, clip_limit=(1.0, 4.0), tile_grid_size=(8, 8))


val: Fast image access ✅ (ping: 1.5±2.0 ms, read: 53.5±12.7 MB/s, size: 131.6 KB)


val: Scanning /content/drive/MyDrive/Colab Notebooks/data/visdrone_raw/val/labels.cache... 548 images, 0 backgrounds, 0 corrupt: 100%|██████████| 548/548 [00:00<?, ?it/s]


Plotting labels to visdrone_yolo11/py-aug-expl/labels.jpg... 
optimizer: AdamW(lr=0.001, momentum=0.8) with parameter groups 106 weight(decay=0.0), 113 weight(decay=0.000375), 112 bias(decay=0.0)
Image sizes 640 train, 640 val
Using 8 dataloader workers
Logging results to visdrone_yolo11/py-aug-expl
Starting training for 50 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       1/50      35.8G      3.766      3.299      1.884       2417        640: 100%|██████████| 136/136 [01:55<00:00,  1.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.84it/s]

                   all        548      38759     0.0258     0.0942     0.0333     0.0136



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       2/50      35.3G      2.654      2.331      1.233       2330        640: 100%|██████████| 136/136 [00:58<00:00,  2.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.60it/s]

                   all        548      38759      0.178      0.108     0.0879     0.0398



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       3/50      38.5G       2.31      1.989      1.147       2083        640: 100%|██████████| 136/136 [00:59<00:00,  2.29it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:04<00:00,  1.32it/s]

                   all        548      38759      0.174      0.156      0.119     0.0582



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       4/50      37.8G      2.121      1.801      1.102       2695        640: 100%|██████████| 136/136 [00:58<00:00,  2.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.56it/s]

                   all        548      38759      0.239      0.193      0.155     0.0776



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       5/50      36.1G      2.012      1.685      1.076       2442        640: 100%|██████████| 136/136 [00:59<00:00,  2.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:04<00:00,  1.50it/s]

                   all        548      38759      0.218      0.199      0.159      0.082



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       6/50      37.9G      1.915      1.588      1.053       2919        640: 100%|██████████| 136/136 [00:58<00:00,  2.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.61it/s]

                   all        548      38759      0.272      0.216      0.181     0.0955



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       7/50      37.6G      1.841      1.513      1.035       2092        640: 100%|██████████| 136/136 [00:59<00:00,  2.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.62it/s]

                   all        548      38759      0.316      0.245      0.221      0.116



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       8/50      37.6G      1.788       1.45      1.021       2986        640: 100%|██████████| 136/136 [00:59<00:00,  2.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.58it/s]

                   all        548      38759      0.305      0.262       0.23      0.127



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       9/50      37.5G      1.744      1.411      1.014       2584        640: 100%|██████████| 136/136 [00:59<00:00,  2.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.59it/s]

                   all        548      38759      0.332      0.268      0.245      0.133



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      10/50      37.4G      1.705       1.37      1.005       2531        640: 100%|██████████| 136/136 [00:58<00:00,  2.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:04<00:00,  1.48it/s]

                   all        548      38759      0.363      0.285      0.265       0.15



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      11/50      38.1G      1.691      1.371      1.002       5505        640:  54%|█████▎    | 73/136 [00:32<00:26,  2.36it/s]

WARNING ⚠️ CUDA OutOfMemoryError in TaskAlignedAssigner, using CPU


      11/50      36.8G      1.685      1.352     0.9986       3015        640: 100%|██████████| 136/136 [01:43<00:00,  1.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.69it/s]

                   all        548      38759      0.385        0.3      0.287       0.16



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      12/50      37.3G       1.65      1.311     0.9921       3012        640: 100%|██████████| 136/136 [00:59<00:00,  2.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.66it/s]

                   all        548      38759      0.377        0.3       0.28      0.157



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      13/50      38.2G      1.637      1.301     0.9871       2393        640: 100%|██████████| 136/136 [00:59<00:00,  2.29it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.64it/s]

                   all        548      38759      0.388      0.315        0.3      0.169



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      14/50      38.3G      1.624      1.285     0.9817       3014        640: 100%|██████████| 136/136 [00:59<00:00,  2.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.64it/s]

                   all        548      38759      0.395      0.321       0.31      0.178



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      15/50      36.6G      1.595      1.255     0.9765       2695        640: 100%|██████████| 136/136 [00:58<00:00,  2.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.67it/s]

                   all        548      38759      0.415       0.32      0.317      0.181



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      16/50      37.8G      1.586      1.246     0.9772       3140        640: 100%|██████████| 136/136 [00:58<00:00,  2.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.68it/s]

                   all        548      38759      0.411       0.32      0.315       0.18



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      17/50      38.7G      1.565      1.223     0.9715       2153        640: 100%|██████████| 136/136 [00:59<00:00,  2.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.66it/s]

                   all        548      38759      0.417      0.326      0.317      0.183



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      18/50      36.5G      1.561      1.214     0.9693       2670        640: 100%|██████████| 136/136 [00:59<00:00,  2.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.68it/s]

                   all        548      38759      0.426      0.331      0.325      0.186



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      19/50      35.2G      1.534      1.191     0.9612       2689        640: 100%|██████████| 136/136 [00:58<00:00,  2.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.66it/s]

                   all        548      38759      0.454      0.333       0.34      0.197



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      20/50      35.5G      1.529      1.186     0.9598       2337        640: 100%|██████████| 136/136 [00:58<00:00,  2.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.66it/s]

                   all        548      38759      0.428      0.338      0.332       0.19



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      21/50      38.4G      1.519      1.172     0.9605       1486        640: 100%|██████████| 136/136 [00:58<00:00,  2.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.66it/s]

                   all        548      38759      0.458      0.342      0.344      0.199



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      22/50      36.7G      1.507      1.156     0.9561       2718        640: 100%|██████████| 136/136 [00:58<00:00,  2.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.63it/s]

                   all        548      38759      0.456      0.352      0.351      0.204



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      23/50      36.7G      1.503      1.157     0.9548       2199        640: 100%|██████████| 136/136 [00:58<00:00,  2.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.62it/s]

                   all        548      38759      0.458      0.355      0.353      0.207



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      24/50      35.5G      1.492      1.141     0.9531       2739        640: 100%|██████████| 136/136 [00:58<00:00,  2.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.66it/s]

                   all        548      38759      0.454      0.357      0.359       0.21



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      25/50      37.2G      1.484      1.138     0.9511       2868        640: 100%|██████████| 136/136 [00:59<00:00,  2.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.64it/s]

                   all        548      38759      0.466      0.357      0.362      0.212



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      26/50      35.9G      1.475      1.128     0.9506       2228        640: 100%|██████████| 136/136 [00:58<00:00,  2.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.66it/s]

                   all        548      38759      0.465      0.361      0.363      0.214



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      27/50      38.1G      1.454      1.104     0.9445       2369        640: 100%|██████████| 136/136 [00:58<00:00,  2.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.68it/s]

                   all        548      38759      0.482      0.369      0.375      0.221



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      28/50      38.2G      1.459       1.11     0.9458       1865        640: 100%|██████████| 136/136 [00:58<00:00,  2.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.66it/s]

                   all        548      38759      0.479      0.366      0.374       0.22



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      29/50      35.6G       1.46      1.107     0.9459       2443        640: 100%|██████████| 136/136 [00:59<00:00,  2.29it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.67it/s]

                   all        548      38759      0.486      0.364      0.371       0.22



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      30/50      38.3G      1.446      1.089     0.9428       2596        640: 100%|██████████| 136/136 [00:58<00:00,  2.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.68it/s]

                   all        548      38759      0.485       0.37      0.378      0.224



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      31/50      35.5G      1.438      1.083       0.94       2471        640: 100%|██████████| 136/136 [00:58<00:00,  2.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.62it/s]

                   all        548      38759      0.495       0.37      0.381      0.226



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      32/50      36.1G      1.425      1.069     0.9366       2561        640: 100%|██████████| 136/136 [00:58<00:00,  2.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.66it/s]

                   all        548      38759       0.49      0.375      0.382      0.226



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      33/50      37.7G      1.417      1.057      0.935       2203        640: 100%|██████████| 136/136 [00:59<00:00,  2.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.62it/s]

                   all        548      38759      0.503      0.373      0.385      0.229



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      34/50      35.9G      1.419      1.061     0.9356       2406        640: 100%|██████████| 136/136 [00:58<00:00,  2.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.65it/s]

                   all        548      38759      0.499      0.375      0.388      0.231



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      35/50      38.5G      1.411      1.052     0.9344       2505        640: 100%|██████████| 136/136 [00:59<00:00,  2.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.66it/s]

                   all        548      38759       0.49      0.382      0.389      0.231



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      36/50      35.6G      1.407      1.049     0.9327       2395        640: 100%|██████████| 136/136 [00:58<00:00,  2.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.65it/s]

                   all        548      38759      0.505      0.382      0.392      0.231



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      37/50      38.3G      1.404      1.039      0.931       2308        640: 100%|██████████| 136/136 [00:58<00:00,  2.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.63it/s]

                   all        548      38759      0.497      0.381      0.391      0.232



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      38/50      35.3G      1.398      1.039     0.9309       2379        640: 100%|██████████| 136/136 [00:58<00:00,  2.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.63it/s]

                   all        548      38759      0.502      0.383      0.393      0.235



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      39/50      37.4G      1.399      1.031     0.9272       6783        640:  27%|██▋       | 37/136 [00:16<00:42,  2.33it/s]

WARNING ⚠️ CUDA OutOfMemoryError in TaskAlignedAssigner, using CPU


      39/50      35.2G      1.402      1.033     0.9282       6015        640:  39%|███▉      | 53/136 [01:04<00:40,  2.06it/s]

WARNING ⚠️ CUDA OutOfMemoryError in TaskAlignedAssigner, using CPU


      39/50      35.2G      1.395      1.031     0.9299       1891        640: 100%|██████████| 136/136 [02:27<00:00,  1.09s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.61it/s]

                   all        548      38759      0.507      0.391      0.398      0.237



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      40/50      34.8G      1.383      1.021      0.926       2731        640: 100%|██████████| 136/136 [00:59<00:00,  2.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.65it/s]

                   all        548      38759      0.507      0.387      0.397      0.237


Closing dataloader mosaic
albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01, method='weighted_average', num_output_channels=3), CLAHE(p=0.01, clip_limit=(1.0, 4.0), tile_grid_size=(8, 8))

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      41/50      35.5G      1.238     0.8972     0.8975        929        640: 100%|██████████| 136/136 [00:57<00:00,  2.38it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.70it/s]

                   all        548      38759      0.496       0.38      0.388      0.232



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      42/50      33.4G      1.219     0.8557     0.8916       1060        640: 100%|██████████| 136/136 [00:54<00:00,  2.52it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.75it/s]

                   all        548      38759        0.5      0.387      0.396      0.237



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      43/50      36.8G      1.213     0.8465     0.8912       1054        640: 100%|██████████| 136/136 [00:54<00:00,  2.51it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.78it/s]

                   all        548      38759      0.508      0.385      0.396      0.237



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      44/50      35.4G      1.206     0.8402     0.8903        846        640: 100%|██████████| 136/136 [00:54<00:00,  2.50it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.63it/s]

                   all        548      38759      0.502      0.392      0.399      0.239



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      45/50      33.2G      1.205      0.834     0.8886       1018        640: 100%|██████████| 136/136 [00:54<00:00,  2.51it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.70it/s]

                   all        548      38759      0.504       0.39      0.399      0.239



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      46/50      36.8G      1.201     0.8304     0.8886        780        640: 100%|██████████| 136/136 [00:54<00:00,  2.51it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.76it/s]

                   all        548      38759      0.513      0.387      0.399      0.239



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      47/50      36.5G      1.198     0.8266     0.8885        850        640: 100%|██████████| 136/136 [00:54<00:00,  2.51it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.74it/s]

                   all        548      38759      0.512      0.389        0.4       0.24



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      48/50      35.8G      1.198     0.8255     0.8876        819        640: 100%|██████████| 136/136 [00:54<00:00,  2.51it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.68it/s]

                   all        548      38759      0.507      0.389        0.4      0.241



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      49/50      36.6G      1.197     0.8254      0.887       1072        640: 100%|██████████| 136/136 [00:54<00:00,  2.51it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.75it/s]

                   all        548      38759      0.509       0.39        0.4      0.241



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      50/50        37G      1.192     0.8225      0.887        848        640: 100%|██████████| 136/136 [00:53<00:00,  2.52it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:03<00:00,  1.72it/s]

                   all        548      38759      0.506       0.39        0.4      0.241



50 epochs completed in 0.928 hours.
Optimizer stripped from visdrone_yolo11/py-aug-expl/weights/last.pt, 40.5MB
Optimizer stripped from visdrone_yolo11/py-aug-expl/weights/best.pt, 40.5MB

Validating visdrone_yolo11/py-aug-expl/weights/best.pt...
Ultralytics 8.3.127 🚀 Python-3.11.12 torch-2.6.0+cu124 CUDA:0 (NVIDIA A100-SXM4-40GB, 40507MiB)
YOLO11m summary (fused): 125 layers, 20,037,742 parameters, 0 gradients, 67.7 GFLOPs


                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:14<00:00,  2.37s/it]


                   all        548      38759      0.486      0.379      0.388      0.233
            pedestrian        520       8844      0.497      0.388      0.405      0.182
                people        482       5125      0.542      0.247      0.306      0.117
               bicycle        364       1287      0.279      0.141      0.117     0.0478
                   car        515      14064       0.67      0.779      0.793      0.554
                   van        421       1975      0.485      0.484      0.463      0.327
                 truck        266        750      0.485      0.395       0.38      0.256
              tricycle        337       1045      0.465       0.27      0.275      0.152
       awning-tricycle        220        532      0.291      0.186      0.165      0.102
                   bus        131        251      0.614       0.53      0.577      0.422
                 motor        485       4886      0.529      0.374        0.4      0.173
Speed: 0.1ms preproce

ultralytics.utils.metrics.DetMetrics object with attributes:

ap_class_index: array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
box: ultralytics.utils.metrics.Metric object
confusion_matrix: <ultralytics.utils.metrics.ConfusionMatrix object at 0x7ef6a838c4d0>
curves: ['Precision-Recall(B)', 'F1-Confidence(B)', 'Precision-Confidence(B)', 'Recall-Confidence(B)']
curves_results: [[array([          0,    0.001001,    0.002002,    0.003003,    0.004004,    0.005005,    0.006006,    0.007007,    0.008008,    0.009009,     0.01001,    0.011011,    0.012012,    0.013013,    0.014014,    0.015015,    0.016016,    0.017017,    0.018018,    0.019019,     0.02002,    0.021021,    0.022022,    0.023023,
          0.024024,    0.025025,    0.026026,    0.027027,    0.028028,    0.029029,     0.03003,    0.031031,    0.032032,    0.033033,    0.034034,    0.035035,    0.036036,    0.037037,    0.038038,    0.039039,     0.04004,    0.041041,    0.042042,    0.043043,    0.044044,    0.045045,    0.046046,    0.0

For this model, we get `mAP50 : 0.388` and `mAP50-95 : 0.233` 

---

## Testing each model

Will use the best.pt for each model and test dataset for final `mAP` evaluations

In [1]:
from ultralytics import YOLO

# Load the best model for each model
py_expl_model = YOLO("content/visdrone_yolo11/py-expl/weights/best.pt")
py_mod_expl_model = YOLO("content/visdrone_yolo11/py-mod-expl/weights/best.pt")
py_aug_expl_model = YOLO("content/visdrone_yolo11/py-aug-expl/weights/best.pt")

In [4]:
py_expl_model.eval()
py_mod_expl_model.eval()
py_aug_expl_model.eval()

# Evaluate each model on the test set
py_expl_results = py_expl_model.val(
    data="visdrone.yaml",
    split="test",
    batch=8,
    imgsz=640,
    device="cuda:0",
    project="visdrone_yolo11",
    name="py-expl-test"
)


Ultralytics 8.3.126  Python-3.10.16 torch-2.5.1+cu121 CUDA:0 (NVIDIA GeForce RTX 4070 Laptop GPU, 8188MiB)
YOLO11m summary (fused): 125 layers, 20,037,742 parameters, 0 gradients, 67.7 GFLOPs
val: Fast image access  (ping: 0.30.1 ms, read: 7.02.6 MB/s, size: 146.5 KB)


val: Scanning C:\Users\manav\Deep Learnig Assignment -2\data\visdrone_raw\test-dev\labels... 1610 images, 0 backgrounds, 0 corrupt: 100%|██████████| 1610/1610 [00:05<00:00, 276.68it/s]


val: New cache created: C:\Users\manav\Deep Learnig Assignment -2\data\visdrone_raw\test-dev\labels.cache


                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 202/202 [00:44<00:00,  4.50it/s]


                   all       1610      75102      0.496      0.388      0.372      0.222
            pedestrian       1197      21006      0.567      0.297      0.324      0.136
                people        797       6376      0.564      0.127      0.173     0.0631
               bicycle        377       1302      0.258      0.185       0.14     0.0617
                   car       1530      28074      0.724      0.746      0.759        0.5
                   van       1168       5771      0.454      0.477      0.426      0.292
                 truck        750       2659      0.496      0.458      0.442      0.293
              tricycle        245        530       0.33      0.353      0.251      0.137
       awning-tricycle        233        599      0.404      0.277      0.231       0.14
                   bus        838       2940      0.686      0.564      0.612      0.443
                 motor        794       5845      0.474      0.399       0.36       0.15
Speed: 0.2ms preproce

For base pre-trained model on test data, we get `mAP50 : 0.372` and `mAP50-95 : 0.222` 

---

In [5]:
py_mod_expl_results = py_mod_expl_model.val(
    data="visdrone.yaml",
    split="test",
    batch=8,
    imgsz=640,
    device="cuda:0",
    project="visdrone_yolo11",
    name="py-mod-expl-test"
)


Ultralytics 8.3.126  Python-3.10.16 torch-2.5.1+cu121 CUDA:0 (NVIDIA GeForce RTX 4070 Laptop GPU, 8188MiB)
YOLO11m summary (fused): 125 layers, 20,037,742 parameters, 0 gradients, 67.7 GFLOPs
val: Fast image access  (ping: 0.10.1 ms, read: 765.6223.8 MB/s, size: 183.1 KB)


val: Scanning C:\Users\manav\Deep Learnig Assignment -2\data\visdrone_raw\test-dev\labels.cache... 1610 images, 0 backgrounds, 0 corrupt: 100%|██████████| 1610/1610 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 202/202 [00:47<00:00,  4.24it/s]


                   all       1610      75102      0.423      0.331      0.309       0.18
            pedestrian       1197      21006      0.473      0.253       0.26      0.103
                people        797       6376      0.482     0.0992      0.134      0.045
               bicycle        377       1302      0.242      0.104     0.0861     0.0352
                   car       1530      28074      0.653      0.717      0.711      0.452
                   van       1168       5771      0.354       0.45      0.369      0.244
                 truck        750       2659      0.416        0.4       0.37      0.243
              tricycle        245        530      0.237      0.245      0.163     0.0882
       awning-tricycle        233        599      0.345      0.195      0.174     0.0953
                   bus        838       2940      0.644      0.519      0.556      0.385
                 motor        794       5845      0.388      0.325      0.272      0.106
Speed: 0.2ms preproce

For backbone modified model on test data, we get `mAP50 : 0.309` and `mAP50-95 : 0.18` 

---

In [6]:
py_aug_expl_results = py_aug_expl_model.val(
    data="visdrone.yaml",
    split="test",
    batch=8,
    imgsz=640,
    device="cuda:0",
    project="visdrone_yolo11",
    name="py-aug-expl-test"
)


Ultralytics 8.3.126  Python-3.10.16 torch-2.5.1+cu121 CUDA:0 (NVIDIA GeForce RTX 4070 Laptop GPU, 8188MiB)
YOLO11m summary (fused): 125 layers, 20,037,742 parameters, 0 gradients, 67.7 GFLOPs
val: Fast image access  (ping: 0.10.0 ms, read: 659.7278.7 MB/s, size: 157.7 KB)


val: Scanning C:\Users\manav\Deep Learnig Assignment -2\data\visdrone_raw\test-dev\labels.cache... 1610 images, 0 backgrounds, 0 corrupt: 100%|██████████| 1610/1610 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 202/202 [00:46<00:00,  4.35it/s]


                   all       1610      75102      0.443       0.34      0.326      0.192
            pedestrian       1197      21006      0.483      0.258      0.267      0.107
                people        797       6376      0.491     0.0997      0.142     0.0485
               bicycle        377       1302      0.225      0.114     0.0841     0.0323
                   car       1530      28074       0.69       0.72       0.73      0.472
                   van       1168       5771      0.375      0.476      0.398      0.268
                 truck        750       2659      0.441      0.426      0.399      0.265
              tricycle        245        530      0.267       0.26      0.186        0.1
       awning-tricycle        233        599      0.383        0.2      0.199      0.113
                   bus        838       2940      0.628      0.535      0.568        0.4
                 motor        794       5845      0.448      0.316      0.288      0.114
Speed: 0.2ms preproce

For base model with training time augmentations, on test data, we get `mAP50 : 0.326` and `mAP50-95 : 0.192` 

---